treeherder/perf/secretary_tool.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from datetime import (datetime,
+                      timedelta)
+
+import simplejson as json
+from django.conf import settings as django_settings
+
+from treeherder.perf.models import (BackfillRecord,
+                                    BackfillReport,
+                                    PerformanceSettings)
+
+logger = logging.getLogger(__name__)
+
+
+class SecretaryTool:
+    """
+    Tool used for doing the secretary work in the Performance Sheriff Bot.
+    """
+    TIME_TO_MATURE = timedelta(hours=4)
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def validate_settings(cls):
+        perf_sheriff_settings, created = PerformanceSettings.objects.get_or_create(
+            name="perf_sheriff_bot",
+            defaults={"settings": cls._get_default_settings()},
+        )
+
+        if created:
+            logger.info("Performance settings for perf_sheriff_bot not found. Creating with defaults.")
+            return
+
+        # reset limits if the settings expired
+        settings = json.loads(perf_sheriff_settings.settings)
+        if cls.are_expired(settings):
+            logger.info(f"Settings are expired. Expired settings: {settings}")
```

Comment:
This is cool, I just learned about formatting with `f` strings.  Thanks!!  :) I've been spending all my time in JavaScript...